feat: add prettier as a code formater for every git commit

hulinpiao opened the pull request:
Added prettier as a code formater for every git commit:
- Config prettier as code formater
- Added .prettierignore file to ignore format checking(including * mark, it play as a swith of turn on/off the format checking)
- Added prettier-format.sh script to handle format fixing and git re-add the fixed files.
- Updated prettier-format.sh script to handle an empty git commit.
- Steup Husky and Lint-staged in order to trigger prettier every git commit command.
- Update README.md

Chris-Wang commented on README.md:
```diff
@@ -96,4 +96,78 @@ query GetUsers {
+Through these steps, every code update is automatically deployed to the development environment and made ready for testing.
+
+### **1. pre-commit checks:** 
+Pre-commit checks are triggered every time a local development task is completed and the code is committed via the command: `git commit`. This job includes code formatting and linting. In this repository, Prettier is used for code formatting, while ESLint is used for linting. Husky and lint-staged are utilized to apply code formatting and linting to files that are staged in Git (via the git add c ...
```
For `pre-commit checks`, technically it's not part of the CI/CD pipeline, as it take place before code is pushed to the repository

Chris-Wang reviewed (changes requested):
Please squash commits into one

Chris-Wang committed:
Added prettier as a code formater for every git commit:

    Config prettier as code formater
    Added .prettierignore file to ignore format checking(including * mark, it play as a swith of turn on/off the format checking)
    Added prettier-format.sh script to handle format fixing and git re-add the fixed files.
    Updated prettier-format.sh script to handle an empty git commit.
    Steup Husky and Lint-staged in order to trigger prettier every git commit command.
    Update README.md
